### PR TITLE
ECO-327: Migrate contract

### PR DESCRIPTION
**Pull request by zie1ony**
# Description
* Move the vesting code to this repository.
* Re-implement tests.

# JIRA
https://casperlabs.atlassian.net/browse/ECO-327

### CODE REVIEW

File: tests/src/lib.rs [hunk 1]
```diff
@@ -61,14 +60,41 @@ mod tests {
...
+	#[test]
+	#[should_panic]
+	fn test_admin_release_by_not_admin() {
+		let mut vesting = VestingContract::deployed();
+		let config: VestingConfig = Default::default();
+		vesting.pause(Sender(ADMIN));
+		vesting.set_block_time(config.admin_release_duration.as_u64());
+		vesting.admin_release(Sender(ALI));
+	}
+
```
Comment (zie1ony) [inline tests/src/lib.rs:92]: Add another test that should panic: `test_withdraw_too_much`.

File: tests/Cargo.toml [hunk 2]
```diff
@@ -7,13 +7,9 @@ edition = "2018"
...
 
 [dependencies]
-casperlabs-contract = "0.4.1"
-casperlabs-types = "0.4.1"
-casperlabs-engine-test-support = "0.6.3"
-casperlabs-engine-core = "0.5.1"
-casperlabs-engine-shared = "0.5.1"
-
-
+casperlabs-contract = { git="https://github.com/zie1ony/CasperLabs", branch="master" }
```
Comment (zie1ony) [inline tests/Cargo.toml:10]: Please changed it everywhere to:
```
{ git="https://github.com/CasperLabs/CasperLabs", branch="hackathon" }
```

File: tests/src/lib.rs [hunk 3]
```diff
@@ -61,14 +60,41 @@ mod tests {
...
-	#[ignore]
 	fn test_admin_release() {
+		let config: VestingConfig = Default::default();
 		let mut vesting = VestingContract::deployed();
 		vesting.pause(Sender(ADMIN));
+		vesting.set_block_time(config.admin_release_duration.as_u64());
 		vesting.admin_release(Sender(ADMIN));
-		let amount = vesting.get_total_amount();
-		assert_eq!(amount, 0);
+		let amount = vesting.get_released_amount();
```
Comment (zie1ony) [inline tests/src/lib.rs:69]: Change types in `VestingConfig` from `U512` to `u64`.

Comment (darthsiroftardis) [reply tests/src/lib.rs:69]: Changing the type seems to break more things, the fix starts affecting the runtime calls. It seems far more convenient to work with U512 and convert to u64 when needed

**Pull request merged**